docs/api.md

```diff
@@ -394,6 +400,35 @@ For more up-to-date documentation, please refer to those docs.
 
     The capabilities to drop.
 
+### `KubernetesHostPathMount`
```

Comment:
Agree with @pwais that using a [persistent volume](https://kubernetes.io/docs/concepts/storage/persistent-volumes/) is probably preferable, if for nothing else than a security reason. It would also generalize better, in case there's a need for volumes that aren't backed by disks on the nodes.

If we do stick with a direct path mount, then we should add a feature flag that the Sematic administrator has to explicitly turn on, as in [this PR](https://github.com/sematic-ai/sematic/pull/751/files#diff-9eddf4dc51bf6a0125ca7fb094468ad284112270385c41cebce4f8f0a29620abR205).

Reply:
> persistent volume

yes, we can recommend this but we can't require it.  if someone needs to access data directly off the host, we need to at least support it even if it's not recommended/ideal.

> feature flag

agreed.

Reply:
Replying to this thread and to [this comment](https://github.com/sematic-ai/sematic/pull/1064#issuecomment-1749547179) by @pwais:

Directly using a "hostPath" is a feature subset of using it while referencing a PersistentVolume spec. Although more robust and extendable in the future, the PersistentVolume route requires the user to create a Kubernetes spec and to apply it.

This feature is actually implemented as a user request for a user that does not want the overhead of also managing PersistentVolumes.

The logic to programmatically create the Pod spec for the "hostPath" will have to be minimally modified in the future to also support referencing a PersistentVolume instead of an arbitrary host path.

So we're going to go ahead with the current version for now.

Added the feature flag.

Reply:
Cool, I think it's reasonable to provide this as a distinct feature.